src/component/component_system_manager.cc

```diff
@@ -120,8 +121,13 @@ tick_doors(ship_space * ship)
     for (auto i = 0u; i < door_man.buffer.num; i++) {
         auto ce = door_man.instance_pool.entity[i];
 
-        /* gas producers require: power, position */
-        assert(switchable_man.exists(ce) || !"gas producer must be switchable");
+        /* doors require: switchable */
+        assert(switchable_man.exists(ce) || !"doors must be switchable");
+
+        /* it's a power door, it's not going /anywhere/ without power */
+        if (!power_man.powered(ce)) {
```

Comment:
Had better assert `power_man.exists(ce)` too, or this will have exciting results.